events/messageReactionAdd.js

```diff
@@ -0,0 +1,45 @@
+const { Event } = require('klasa');
+
+module.exports = class extends Event {
+
+    constructor(...args) {
+        super(...args, { 
+            enabled: true
+        });
+    }
+
+    run(messageReaction, user) {
+
+        if(!messageReaction.message.guild.configs.starboard || !messageReaction.message.guild.configs.starboardCount) return;
+        if(messageReaction.message.author.id === user.id) {
+            messageReaction.message.reply('You cannot star your own messages.')
+            .then(messageReaction.users.remove(user.id))
+            .then(message => setTimeout(() => message.delete(), 5000))
+            return;
+        }
+
+        const avatarOptions = messageReaction.message.author.displayAvatarURL().endsWith('.gif') ? { format: 'gif' } : { format: 'webp' }
+
+        const embed = new this.client.methods.Embed()
+        .setAuthor(messageReaction.message.member.displayName + messageReaction.message.author.discriminator, messageReaction.message.author.displayAvatarURL(avatarOptions))
+        .setDescription(messageReaction.message.content)
+        .setTimestamp()
+        .setFooter(`${messageReaction.message.id}`)
+
+        if(!messageReaction.message.attachments.length == 0) {

```

Comment:
Dont use `==`, also attachments is a Collection so `messageReaction.message.attachments.size`, no need for `!` or `== 0`